### ISSUE TITLE
Incorrect intersection type error

### ISSUE THREAD

**Issue by haxiomic** [HaxeFoundation/haxe#9455]
In haxe 4.0.5 this compiles fine, in 4.1.0 (including latest commit 354c24d308f1781e27bbc830edeaf5d5b1d465bd), it fails with 

`Main.hx:1: characters 24-38 : Can only extend structures`

**Main.hx**
```haxe
typedef Intersection = VueConstructor & {};

typedef VueConstructor = {
	@:overload(function(?options:Intersection):Void { })
	function extend():Void;
};

class Main {
    static function main() {}
}
```

Discovered by @muchitto after converting vue.js externs (this is a reduced example). See https://github.com/haxiomic/dts2hx/issues/11#issuecomment-631593956

Removing the @:overload allows it to compile

I guess dts2hx is at least useful as a fuzzer for the compiler :P

**Comment by RealyUniqueName** [comment]
This happens with the released 4.1.0 even without latest development changes.

**Comment by haxiomic** [comment]
Thanks @RealyUniqueName! :)
